src/Castle.Facilities.AspNet.Mvc/AspNetMvcControllerFactory.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright 2004-2017 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Reflection;
+
+namespace Castle.Facilities.AspNet.Mvc
+{
+	using System;
+	using System.Web.Mvc;
+	using System.Web.Routing;
+	using System.Web.SessionState;
+
+	using Castle.Facilities.AspNet.Mvc.Exceptions;
+	using Castle.MicroKernel;
+
+	public class AspNetMvcControllerFactory : IControllerFactory
+	{
+		private readonly IKernel container;
+		private Assembly controllerAssembly;
```

Comment:
> A single controllerAssembly seems wrong, doesn't ASP.NET MVC support controllers in as many assemblies as you like? I also don't really like that we are duplicating MVC functionality like appending "Controller", and not caching controller type lookups which MVC already does.

Good spot, it is now a list and you can chain them together to take advantage of the fluent interface.

> I assume you considered inheriting from DefaultControllerFactory, overriding GetControllerInstance and ReleaseController, then using GetControllerType from the base? What went wrong with that option?

Don't need any of that base behaviour.

Reply:
Happy to take your lead on this because I'm not using MVC, but I still would have thought not having to configure your controller assemblies would be a nicer contract and be more convention over configuration like.